cpp/src/io/parquet/reader_impl_helpers.cpp

```diff
@@ -962,6 +972,62 @@ aggregate_reader_metadata::get_rowgroup_metadata() const
   return rg_metadata;
 }
 
+std::unordered_map<std::string, std::vector<int64_t>>
+aggregate_reader_metadata::get_column_chunk_metadata() const
+{
+  std::unordered_map<std::string, std::vector<int64_t>> column_chunk_metadata;
+
+  std::function<void(std::string const&, int)> const populate_column_chunk_metadata =
```

Comment:
Do we have to use a std::function rather than a lambda here to allow for recursion?

Reply:
`std::function<>` is needed to to fully capture itself as it involves recursion.